### PR TITLE
AST-1618 - Global typography section takes time to expand

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- Due to number of controls inside, the Global > Typography control take time
- Revamped open typo controls into sub-controls of settings-group to make it lazy load
- Due to that implementation of settings-group, typo controls will lazily load 
- Migrated variant control from PHP to React.

### Screenshots
- https://share.bsf.io/4guKW8YQ
- Speed - https://share.bsf.io/bLuKbgze

### Types of changes
- Bug fix (non-breaking change which fixes an issue) 

### How has this been tested?
- Check master version - Global > Typography section (12 sec, local dev setup)
- In current version - Global > Typography section (1 sec, same local setup)
- Check variant control backward, new setup, different fonts etc

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/customizer/class-astra-font-families.php [hunk 1]
```diff
@@ -117,6 +117,36 @@ public static function get_custom_fonts() {
...
+				'500'       => __( 'Medium 500', 'astra' ),
+				'600'       => __( 'Semi-Bold 600', 'astra' ),
+				'700'       => __( 'Bold 700', 'astra' ),
+				'800'       => __( 'Extra-Bold 800', 'astra' ),
+				'900'       => __( 'Ultra-Bold 900', 'astra' ),
+				'100italic' => __( 'Thin 100 Italic', 'astra' ),
+				'200italic' => __( 'Extra Light 200 Italic', 'astra' ),
+				'300italic' => __( 'Light 300 Italic', 'astra' ),
+				'400italic' => __( 'Regular 400 Italic', 'astra' ),
+				'italic'    => __( 'Regular 400 Italic', 'astra' ),
```
Comment (nileshc-bsf) [inline inc/customizer/class-astra-font-families.php:141]: @imnavanath - Seems to be repeated i.e. 400italic and italic, not sure, is this intentional.

**Review by nileshc-bsf — APPROVED** [review]

File: inc/customizer/class-astra-font-families.php (hunk 1 above)
Comment (imnavanath) [reply inc/customizer/class-astra-font-families.php:141]: Yes the label is repeated, the case is: Some fonts have variants `400italic` & some are having `italic` but its one the same `Regular 400 Italic` .

Comment (nileshc-bsf) [reply inc/customizer/class-astra-font-families.php:141]: Yes, Got your point, thanks for the clarification 👍

**Review by nileshc-bsf — COMMENTED** [review]
Add the Changelog as well for this typography improvement.

File: inc/customizer/custom-controls/assets/css/unminified/custom-controls-rtl.css [hunk 2]
```diff
@@ -3419,6 +3421,9 @@ input[type="text"].select2-search__field {
 .ast-responsive-select-container.active {
   display: block;
 }
+.customize-control select, .ast-variant-select > div, .customize-control input {
+  border-color: #ddd;
```
Comment (premanshup) [inline inc/customizer/custom-controls/assets/css/unminified/custom-controls-rtl.css:3425]:
```suggestion
  border-color: #dddddd;
```

File: inc/customizer/extend-custom-controls/src/ast-font-variant/ast-font-variant.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		help,
+		label,
+		name,
+		variant,
+	} = props.control.params;
+
+	const [ propValue, setValue ] = useState( props.control.setting.get() || [] );
+	const [ fontVal, setfontVal ] = useState( wp.customize.control( variant ).setting.get() || 'inherit' );
+
+	var controlCleanName = props.control.params.name;
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/ast-font-variant/ast-font-variant.js:18]:
```suggestion
	let controlCleanName = props.control.params.name;
```

**Review by premanshup — APPROVED** [review]

File: inc/customizer/extend-custom-controls/src/ast-font-variant/ast-font-variant.js (hunk 3 above)
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/ast-font-variant/ast-font-variant.js]: Done

File: inc/customizer/custom-controls/assets/css/unminified/custom-controls-rtl.css (hunk 2 above)
Comment (imnavanath) [reply inc/customizer/custom-controls/assets/css/unminified/custom-controls-rtl.css]: Done

**Pull request merged**